src/hr/hr.controller.ts

```diff
@@ -0,0 +1,15 @@
+import { Controller, Body, HttpStatus, HttpCode, Patch } from '@nestjs/common';
+import { HrCreateRequest, HrCreatedResponse } from 'src/types';
+
+import { HrService } from './hr.service';
+
+@Controller('hr')
+export class HrController {
+  constructor(private readonly hrService: HrService) {}
+
+  @HttpCode(HttpStatus.OK)
+  @Patch('/')
+  createHr(@Body() hrData: HrCreateRequest): Promise<HrCreatedResponse> {
```

Comment:
tutaj powinieneś mieć `hrData: CreateHrDto`, żeby to miało sens i działało.

Reply:
Działa z HrCreateRequest. Nie rozumiem dlaczego użycie interfejsu nie ma sensu. Zmienione na DTO.

Reply:
W DTO dodajesz walidacje :)